src/extensionInstall.mts

```diff
@@ -0,0 +1,71 @@
+import electron, { ProgressReporter } from '@vscode/test-electron';
+import { spawnSync } from 'node:child_process';
+import { createRequire } from 'node:module';
+import path from 'node:path';
+
+const require = createRequire(import.meta.url);
+
+export async function installExtensions(
+  extensionDevelopmentPath: string | string[],
+  extensions: string[] = [],
+  installDependentExtensions: boolean = false,
+  codeVersion?: string,
+  desktopPlatform?: string,
+  reporter?: ProgressReporter,
+) {
+  const vscodePath = await electron.downloadAndUnzipVSCode(codeVersion, desktopPlatform, reporter);
+
+  const extensionDevelopmentPaths = Array.isArray(extensionDevelopmentPath)
+    ? extensionDevelopmentPath
+    : [extensionDevelopmentPath];
+
+  const extensionsToInstall = installDependentExtensions
+    ? mergeDependentExtensions(extensions, extensionDevelopmentPaths)
+    : extensions;
+
+  const [cli, ...cliArgs] = electron.resolveCliArgsFromVSCodeExecutablePath(vscodePath, {
+    platform: desktopPlatform,
+  });
+
+  for (const extension of extensionsToInstall) {
+    cliArgs.push('--install-extension', extension);
+  }
+
+  console.debug('Installing VSCode Extensions', cli, cliArgs);
+
+  // TODO: Start async and stream the results
+  const installResult = spawnSync(cli, cliArgs, { encoding: 'utf-8' });
+  if (installResult.stderr) {
```

Comment:
We should only fail if the exit code != 0

Reply:
So we should not fail and go on with tests even if extensions didn't install? I'm pretty sure it did exit code 0 when the extension didn't install hence this check, I will reverify.

Reply:
Confirmed that it exits status 0 but still stderr, so I'm going to currently leave this as is unless it is fixed in vscode proper.
![image](https://github.com/microsoft/vscode-test-cli/assets/15258962/76e92704-dc2e-4512-a0af-49721751d7ed)

Reply:
Issue filed: https://github.com/microsoft/vscode/issues/204067